pyhf/__init__.py

```diff
@@ -38,18 +39,18 @@ def set_backend(backend):
         None
 
     Example:
-        pyhf.set_backend(tensorflow_backend(session=tf.Session()))
+        pyhf.set_backend(pyhf.tensor.tensorflow_backend(session=tf.Session()))
     """
     global tensorlib
     global optimizer
 
     tensorlib = backend
     if isinstance(tensorlib, tensor.tensorflow_backend):
         optimizer = optimize.tflow_optimizer(tensorlib)
-    elif isinstance(tensorlib,tensor.pytorch_backend):
+    elif isinstance(tensorlib, tensor.pytorch_backend):
         optimizer = optimize.pytorch_optimizer(tensorlib=tensorlib)
     # TODO: Add support for mxnet_optimizer()
-    # elif isinstance(tensorlib, mxnet_backend):
+    # elif isinstance(tensorlib, tensor.mxnet_backend):
```

Comment:
should say `pyhf.tensor.mxnet_backend`

Reply:
Well, a few lines above and below it, you use `pyhf.tensor....` so just be consistent.

Reply:
No, the only place `pyhf.tensor` appears in the file is in the import and in the doc string example:
```python
    """
    Set the backend and the associated optimizer

    Args:
        backend: One of the supported pyhf backends: NumPy,
                 TensorFlow, PyTorch, and MXNet

    Returns:
        None

    Example:
        pyhf.set_backend(pyhf.tensor.tensorflow_backend(session=tf.Session()))
    """
```
It appears in the doc string as I am not making assumptions on what someone has imported. Do you want me to change the doc string instead of the import?

Reply:
Yes, that would be better.

Reply:
So you want something like
```
    Example:
        import pyhf.tensor as tensor
        pyhf.set_backend(tensor.tensorflow_backend(session=tf.Session()))
    """
```
?
It seems to me that doc string example really should not make any assumptions as it will most likely not be ready by someone reading the code, but rather in docs or in the suggestion/info box of an editor. I'm missing why this is better, so can you give an example?

Reply:
> It seems to me that doc string example really should not make any assumptions as it will most likely not be ready by someone reading the code, but rather in docs or in the suggestion/info box of an editor. I'm missing why this is better, so can you give an example?

Yes. At some point, the docstrings will need to get tested, so if we just start writing good examples now and clean up the structure later -- that makes it easier.

Reply:
Ah, now that I understand that for doctest each doc-block is tested in a new python shell with only the code `pip install -e` the point makes sense. Thanks for pointing this out!

Reply:
Fixed in bdc74a4